src/scripts/classname.ts

```diff
@@ -0,0 +1,19 @@
+type Rule = {
+  [key: string]: boolean;
+};
+
+type Input = Array<string | Rule>;
+
+export const classname = (...rules: Input): string => (
```

Comment:
impressive job man 🔥 
But I don't see you have the constant classes covered here. I mean, classnames should also have the possibility to add styles that don't need condition. We can do it like: classnames
```
cn('some-class ml-8 another-class', {
   'has-background-success-light': selectedGood === good 
})
```
where in this case the classnames `some-class ml-8 another-class` will be applied anyway, and the class 'has-background-success-light' only if the condition is true

Reply:
I actually have it here 😊 if a rule is only a string just pastes it to the list.
That's why the input can be string too.

```ts
    if (typeof rule === 'string') {
      return `${classes} ${rule}`;
    }
 ```
 
thanks a lot, happy to read these